src/Npgsql/NpgsqlWriteBuffer.cs

```diff
@@ -448,7 +448,7 @@ public Stream GetStream()
         internal void WriteStringChunked(char[] chars, int charIndex, int charCount,
                                          bool flush, out int charsUsed, out bool completed)
         {
-            if (WriteSpaceLeft == 0)
+            if (WriteSpaceLeft < _textEncoder.GetByteCount(chars, charIndex, 1, true))
```

Comment:
Unless I'm mistaken, that `true` should instead instead be `false` - we shouldn't flush the internal state of the encoder as we're definitely going to use it after this call (regardless of whether we return here or not).

A super-small nit: we tend to now use explicit parameter naming for bool literals, so `flush: false`.

Reply:
Addressed in 06435296f48b459e2c13ffcb2640284ce1b4c46e, thanks @roji!